app/models/media_resource.rb

```diff
@@ -4,10 +4,36 @@ class MediaResource < ApplicationRecord
 
   belongs_to :episode, -> { with_deleted }, touch: true, optional: true
 
+  acts_as_paranoid
+
   enum status: [:started, :created, :processing, :complete, :error, :retrying, :cancelled]
 
   before_validation :initialize_attributes, on: :create
 
+  after_create :replace_resources!
+
+  scope :complete_or_replaced, -> do
+    with_deleted
+      .complete
+      .where("deleted_at IS NULL OR replaced_at IS NOT NULL")
+      .order("created_at DESC")
+  end
+
+  def self.build(file, position = nil)
```

Comment:
Does this need to override the rails default build semantics to do what you want?  This could be a helper method `build_for_replacement`. ?